src/OrleansProviders/Streams/Memory/MemoryMessageData.cs

```diff
@@ -0,0 +1,39 @@
+﻿
+using System;
+
+namespace Orleans.Providers
+{
+    /// <summary>
+    /// Represents the event sent and received from an In-Memory queue grain. 
+    /// </summary>
+    [Serializable]
+    public struct MemoryMessageData
+    {
+        /// <summary>
+        /// Stream Guid of the event data.
+        /// </summary>
+        public Guid StreamGuid;
+        /// <summary>
+        /// Hash code of namespace. Stores hash for fast and efficient storage.
+        /// </summary>
+        public string StreamNamespace;
```

Comment:
Comment updated.
Going to stick with string rather than hash, because it would complicate the serialization story and performance is less of a concern for this provider as it's intended for test.  If/when GC because an issues, we will address it.